src/CHIPBindings.cc

```diff
@@ -630,7 +694,12 @@ hipError_t hipGraphKernelNodeGetParams(hipGraphNode_t node,
                                        hipKernelNodeParams *pNodeParams) {
   CHIP_TRY
   CHIPInitialize();
-  *pNodeParams = ((CHIPGraphNodeKernel *)node)->getParams();
+  if (!node || !pNodeParams)
+    RETURN(hipErrorInvalidValue);
+  auto CastNode = static_cast<CHIPGraphNodeKernel *>(node);
```

Comment:
we should probably check the result of the cast?

Reply:
Static cast is not a runtime check and the check results in undefined behavior (UB) if the `node` is not a subclass of `hipGraphNode_t`. Such case to occur probably means that the user does something which results in UB before the call.